data-azure-cosmos/src/test/groovy/io/micronaut/data/azure/NoPartitionKeyCosmosDbSpec.groovy

```diff
@@ -178,4 +188,28 @@ class NoPartitionKeyCosmosDbSpec extends Specification implements AzureCosmosTes
             def ex = thrown(IllegalStateException)
             ex.message.contains("unsupported identity type")
     }
+
+    @Requires(property = "spec.name", value = "NoPartitionKeyCosmosDbSpec")
+    @Singleton
+    static class RequestChargeCosmosDiagnosticsProcessor implements CosmosDiagnosticsProcessor {
+
+        private double requestChargeSum
```

Comment:
I would use `AtomicInteger`

Reply:
It was summing double values, but since test checks just if > 0 then it makes sense, updated.